sample/src/main/java/com/adevinta/android/barista/sample/CameraActivity.kt

```diff
@@ -1,46 +1,79 @@
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat
 import androidx.core.content.FileProvider
 import com.bumptech.glide.Glide
-import kotlinx.android.synthetic.main.activity_camera.image_view
-import kotlinx.android.synthetic.main.activity_camera.take_picture
 import java.io.File
 
-class CameraActivity : AppCompatActivity() {
-
-  companion object {
-    val TAKE_PICTURE_REQUEST_CODE = 42
-  }
-
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_camera)
-    take_picture.setOnClickListener {
-      val uri = getPictureUri()
-      val intent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
-      intent.putExtra(MediaStore.EXTRA_OUTPUT, uri)
-      intent.putExtra("return-data", false)
-      startActivityForResult(intent, TAKE_PICTURE_REQUEST_CODE)
+class CameraActivity : ComponentActivity() {
+
+    lateinit var button: Button
+    lateinit var imageView: ImageView
+
+    private val capturePhoto = registerForActivityResult(ActivityResultContracts.TakePicture()) {
+        if (it) {
+            Glide.with(this).load(getPictureUri()).into(imageView)
+        }
     }
-  }
 
-  @SuppressLint("MissingSuperCall")
-  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-    if (requestCode == TAKE_PICTURE_REQUEST_CODE) {
-      Glide.with(this).load(getPictureUri()).into(image_view)
+    private val requestPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestPermission()
+        ) { isGranted: Boolean ->
+            if (isGranted) {
+                capturePhoto.launch(getPictureUri())
+            } else {
+                Toast.makeText(
+                    this,
+                    "You need to enable access to camera from device settings",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_camera)
+        imageView = findViewById(R.id.image_view)
+        button = findViewById(R.id.take_picture)
+
+        button.setOnClickListener {
+            if (ContextCompat.checkSelfPermission(
+                    this,
+                    Manifest.permission.CAMERA
+                ) == PackageManager.PERMISSION_GRANTED
+            ) {
+                capturePhoto.launch(getPictureUri())
+            }
+            else {
+                // You can directly ask for the permission.
+                // The registered ActivityResultCallback gets the result of this request.
+                requestPermissionLauncher.launch(
```

Comment:
Do we really need this change? In the previous version, if the user didn't give the permission, the camera launch call broke, and that's great to capture it from the test side. WDYT? It's just to avoid messing with something it worked. I get this new version lets you, as a user, interact with this Activity, but that's not supposed to happen. This Activity exists only to check Barista permissions handler works as expected.

Reply:
Ah ok. I didn't have much context of what the purpose of the CameraActivity is. However, I assumed it was broken and I guess this is intentional? If so I can revert the changes to keep the PR minimal.

Reply:
You are totally right. We should explain it in the contributors page. I opened an issue to remember it: https://github.com/AdevintaSpain/Barista/issues/455

Sorry for the invested time we're going to waste! 😞

Reply:
No worries at all. Thanks for clarifying.